core/types/block.libevm.go

```diff
@@ -27,40 +28,50 @@ import (
 // HeaderHooks are required for all types registered with [RegisterExtras] for
 // [Header] payloads.
 type HeaderHooks interface {
+	MarshalJSON(*Header) ([]byte, error) //nolint:govet // Type-specific override hook
+	UnmarshalJSON(*Header, []byte) error //nolint:govet
```

Comment:
nit Would it be possible to rename those for example to `MarshalJSONWithHeader` to calm govet?

Reply:
I don't think we should be making code worse (the `WithHeader` is redundant because it's communicated by the argument) to placate an overly zealous linter, particularly when we can disable it.